proof.go

```diff
@@ -206,10 +206,18 @@ func (proof Proof) verifyLeafHashes(nth *Hasher, verifyCompleteness bool, nID na
 	return bytes.Equal(tree.Root(), root)
 }
 
-func (proof Proof) VerifyInclusion(h hash.Hash, nid namespace.ID, data []byte, root []byte) bool {
+func (proof Proof) VerifyInclusion(h hash.Hash, nid namespace.ID, data [][]byte, root []byte) bool {
```

Comment:
This api is arbitrarily restricting proofs to the same namespace. While that fits our use case for the rest of the repos, the api could be simplified to match `tree.Push()` where the namespace is included in the data. This would simplify some other things as well, but I felt that was outside the scope of this PR.

Reply:
Can you document this public function? I don't actually know what `[][]byte` is just by looking at it.

Reply:
sure thing https://github.com/celestiaorg/nmt/pull/58/commits/d40c9b5eb2eaa19f5307663d9b40b1622221248c